draft-ietf-tls-esni.md

```diff
@@ -719,10 +719,17 @@ Upon receiving an "encrypted_server_name" extension, the client-facing
 server MUST check that it is able to negotiate TLS 1.3 or greater. If not,
 it MUST abort the connection with a "handshake_failure" alert.
 
-If the ClientEncryptedSNI.record_digest value does not match the
-cryptographic hash of any known ESNIKeys structure, it MUST ignore the
-extension and proceed with the connection, with the following added
-behavior:
+The ClientEncryptedSNI value is said to match a known ESNIKeys if there exists
+an ESNIKeys that can be used to successfully decrypt ClientEncryptedSNI.encrypted_sni.
+This matching procedure should be performed as follows. If ClientEncryptedSNI.record_digest
+is non-empty, servers SHOULD compare it against cryptographic hashes of known ESNIKeys and
+choose the one that matches. Otherwise, if ClientEncryptedSNI.record_digest is empty,
+servers MAY use trial decryption to match to a known ESNIKeys. If both checks fail,
+there is no matching ESNIKeys.
```

Comment:
This "otherwise" seems to forbid random record_digests (as opposed to empty) or to fall back to trial decryption.. Is that intended?

Reply:
Hmm, good point. It's not intended. I'll remove "Otherwise."